File: app/graphql/mutations/create_activity.rb
module Mutations
  class CreateActivity < BaseMutation
    description 'Create an activity'
    payload_type Types::Activity
    null false

    argument :year, ID, required: true
    argument :type, Types::ActivityType, required: true
    argument :attributes, Types::ActivityAttributes, required: true

    def resolve(year:, type:, attributes:)
      ::CreateActivity.call(
        current_user: current_user,
        festival: festival_by_year(year),
        attributes: attributes.to_h.merge(type: type)
      ).activity
    end
  end
end

File: app/javascript/queries/mutations/create_activity.js
import gql from 'graphql-tag'

export default gql`
  mutation CreateActivity($year: Int!, $type: ActivityType!, $attributes: ActivityAttributes!) {
    createActivity(year: $year, type: $type, attributes: $attributes) {
      id
      name
      type
    }
  }
`
